tools/yaml-templates/web-e2e-tests-job.yml

```diff
@@ -0,0 +1,73 @@
+# This template handles running E2E tests (including all necessary setup steps) from the host SDK repo for various environments.
+# It will run one E2E job for each testPrefixPattern selector in the testPrefixPatternGroups parameter.
+
+parameters:
+  - name: AppHostingSdk
+    default: none
+    type: string
+  - name: hostingEnvironmentType
+    type: string
+    values:
+      - 'electron'
+      - 'standardWeb'
+  - name: teamsJsReferenceType
+    type: string
+    values:
+      - 'npm'
+      - 'scriptTag'
+  - name: testPrefixPatternGroups
+    type: object
+    default: ['{[A-Z],[a-z],[0-9]}']
+
+jobs:
+  - ${{each testPrefixPattern in parameters.testPrefixPatternGroups}}:
+      - job: E2ETestsWeb_${{parameters.teamsJsReferenceType}}_${{ replace(replace(replace(replace(replace(replace(testPrefixPattern, '[', '_'), ']', '_' ), '{', '_'), '}', '_'), ',', '_'), '-', '_') }}
```

Comment:
>replace(replace(replace(replace(replace(replace [](http://example.com/codeflow?start=66&length=47)

I know this looks gross but I don't know of any other way to do it:
- Jobs must have names
- Each job must have a unique name, so the job name must include some data from the parameters that identifies what makes it unique
- Job names can only have alphanumerics and underscores
- This replace function doesn't support RegEx or character groups, so I can't provide a pattern like `[^A-za-z] ` etc.
- Consequently, I have to individually replace each expected "illegal" character with the underscore.
- Job names also can't span multiple lines so this is as readable as I can make it

Reply:
Consider putting a comment above the line indicating what this is doing? I think it's basically replacing all illegal characters with underscore to use as a valid job name.